runners/core-java/src/main/java/org/apache/beam/runners/core/TimerInternals.java

```diff
@@ -217,6 +218,7 @@ public int compareTo(TimerData that) {
         // Obtaining the stringKey may be expensive; only do so if required
         chain = chain.compare(getNamespace().stringKey(), that.getNamespace().stringKey());
       }
+      chain = chain.compare(this.getTimerId(), that.getTimerId());
```

Comment:
I'd move this up to before the namespace, and just tack it on to the rest of the chain - it's cheap to retrieve the strings, and they won't be compared if we don't need to compare them.

Reply:
Done.